integration-tests/common.go

```diff
@@ -2,77 +2,173 @@ package integration_tests
-	return urls, nil
+type StarkNetDevnetClient struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	cfg    *blockchain.EVMNetwork
+	client *resty.Client
 }
 
-func NewStarkNetClient(cfg *StarkNetNetworkConfig, urls []*url.URL) (*StarkNetClient, error) {
+type ChainlinkClient struct {
+	nKeys          []ctfClient.NodeKeysBundle
+	chainlinkNodes []*client.Chainlink
+	bTypeAttr      *client.BridgeTypeAttributes
+	bootstrapPeers []client.P2PData
+}
+
+func (t *Test) NewStarkNetDevnetClient(cfg *blockchain.EVMNetwork) (*StarkNetDevnetClient, error) {
 	ctx, cancel := context.WithCancel(context.Background())
-	c := &StarkNetClient{
+	t.sc = &StarkNetDevnetClient{
 		ctx:    ctx,
 		cancel: cancel,
 		cfg:    cfg,
-		urls:   urls,
-		client: resty.New().SetBaseURL(urls[0].String()),
+		client: resty.New().SetBaseURL(t.GetStarkNetAddress()),
+	}
+	// Currently not needed since we are testing on L2
+	// if err := c.init(); err != nil {
+	// 	return nil, err
+	// }
+	return t.sc, nil
+}
+
+// Deploys and sets up config of the environment and nodes
+func (t *Test) DeployCluster(nodes int) *Test {
+	t = &Test{
+		sc: &StarkNetDevnetClient{},
+		cc: &ChainlinkClient{},
 	}
-	if err := c.init(); err != nil {
-		return nil, err
+	t.DeployEnv(nodes)
+	t.SetupClients()
+	t.CreateKeys()
+	return t
+}
+
+// Deploys the environment
+func (t *Test) DeployEnv(nodes int) {
+	t.Env = environment.New(&environment.Config{
+		NamespacePrefix: "smoke-ocr-starknet",
+		TTL:             3 * time.Hour,
+		InsideK8s:       false,
+	}).
+		// AddHelm(hardhat.New(nil)).
+		AddHelm(ops.New(nil)).
+		AddHelm(mockservercfg.New(nil)).
+		AddHelm(mockserver.New(nil)).
+		AddHelm(chainlink.New(0, map[string]interface{}{
+			"replicas": nodes,
+			"chainlink": map[string]interface{}{
+				"image": map[string]interface{}{
+					"image":   "795953128386.dkr.ecr.us-west-2.amazonaws.com/chainlink",
+					"version": "custom.6e31f3079442a3c50980a4929110aff64489f4cd",
+				},
+			},
+			"env": map[string]interface{}{
+				"STARKNET_ENABLED":            "true",
+				"EVM_ENABLED":                 "false",
+				"EVM_RPC_ENABLED":             "false",
+				"CHAINLINK_DEV":               "false",
+				"FEATURE_OFFCHAIN_REPORTING2": "true",
+				"feature_offchain_reporting":  "false",
+				"P2P_NETWORKING_STACK":        "V2",
+				"P2PV2_LISTEN_ADDRESSES":      "0.0.0.0:8090",
+				"P2PV2_DELTA_DIAL":            "5s",
+				"P2PV2_DELTA_RECONCILE":       "5s",
+				"p2p_listen_port":             "0",
+			},
+		}))
+	err := t.Env.Run()
+	Expect(err).ShouldNot(HaveOccurred())
+	t.mockServer, err = ctfClient.ConnectMockServer(t.Env)
+	Expect(err).ShouldNot(HaveOccurred(), "Creating mockserver clients shouldn't fail")
+
+}
+
+// Sets up the starknet client
+func (t *Test) SetupClients() {
+	t.sc, err = t.NewStarkNetDevnetClient(&blockchain.EVMNetwork{
+		Name: "starknet-dev",
+		URL:  t.Env.URLs[devnetClient][1],
+		PrivateKeys: []string{
+			"c4da537c1651ddae44867db30d67b366",
```

Comment:
Also redeclared in `smoke/ocr2_test.go`, consider extracting this as a specific env (default) somewhere.

Reply:
Moved to `starknet` package and set default values as well as getters.